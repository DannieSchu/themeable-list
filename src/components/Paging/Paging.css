.Paging {
  display: flex;
  justify-content: center;
  margin-top: 0;

  button {
    border-radius: 100%;  
    padding: 0.5rem 0.6rem;
    margin: .5rem;
    text-align: center;

    background-color: skyblue;
    cursor: pointer;
    box-shadow: inset 0 0 6px rgb(207, 205, 211);

  }
  
  button:active {
    background-color: rgb(9, 49, 97);
    color: skyblue;
    box-shadow: none;

  }
  
  button:disabled {
    background-color: rgb(238, 202, 192);

    cursor: default
  }
}

