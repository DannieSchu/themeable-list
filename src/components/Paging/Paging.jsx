import React from 'react';
import PropTypes from 'prop-types';

const Paging = ({ incrementPage, page }) => {
  return (
    <div>
      <button onClick={() => incrementPage(-1)} disabled={page === 1}>&lt;</button>
      <button onClick={() => incrementPage(1)} disabled={page === 4}>&gt;</button>
    </div>
  );
};

Paging.propTypes = {
  incrementPage: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired
};

export default Paging;

