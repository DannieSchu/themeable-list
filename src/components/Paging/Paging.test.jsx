import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(<Paging incrementPage={() => {}} page={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});

