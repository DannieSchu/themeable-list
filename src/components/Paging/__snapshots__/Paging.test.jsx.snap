// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<div>
  <button
    disabled={true}
    onClick={[Function]}
  >
    &lt;
  </button>
  <button
    disabled={false}
    onClick={[Function]}
  >
    &gt;
  </button>
</div>
`;
